detox/src/devices/drivers/android/tools/FreeDeviceFinder.js

```diff
@@ -11,28 +11,38 @@ class FreeDeviceFinder {
   async findFreeDevice(deviceQuery) {
     const { devices } = await this.adb.devices();
     for (const candidate of devices) {
-      const isBusy = this.deviceRegistry.isDeviceBusy(candidate.adbName);
-      if (isBusy) {
-        log.debug({ event: DEVICE_LOOKUP_LOG_EVT }, `Device ${candidate.adbName} is busy, skipping...`);
-        continue;
+      if (await this._isDeviceFreeAndMatching(candidate, deviceQuery)) {
+        log.debug({ event: DEVICE_LOOKUP_LOG_EVT }, `Found a matching & free device ${candidate.adbName}`);
```

Comment:
Can't we have all those `log.debug` calls inside `_isDeviceFreeAndMatching` function?

Reply:
Yes, we can!